### ISSUE TITLE
Allow separate configuration of runtime and config state for SELinux

### ISSUE THREAD

**Issue by lfrancke** [ansible/ansible#6520]
##### Issue Type:

Feature Idea
##### Ansible Version:

ansible 1.5.3
##### Environment:

N/A
##### Summary:

Disabling SELinux takes a full reboot. Until that can happen permissive mode is often good enough.
##### Steps To Reproduce:

Currently setting state to disabled doesn't change the runtime state at all but (this might be unintended) it always reports as changed regardless of the current config state.

Setting state to permissive changes runtime and config state.

What I'd like is either:
- When setting state to disabled automatically set runtime state to permissive (maybe make this an option?)
- Allow separate configuration of runtime and config state

It should be able to work around the problem by grepping the sysctl file and the sestatus and then depending on the result running the selinux module twice. Once to set it to permissive and once more afterwards to set it to disabled. But only do that when the current runtime state is not already disabled. I haven't tried this but I think it'd work. I think one of the other options would be much cleaner though.
##### Expected Results:

N/A
##### Actual Results:

N/A

**Comment by mpdehaan** [comment]
This will not be an option as changing SELinux to permissive means it is non-functional, which is (what I hope would be) completely undesirable to most every user.

However, should you wish to do this you could make a "shell" module call to make it hapepn.

**Comment by lfrancke** [comment]
I don't quite understand your comment.

I am trying to make SELinux non-functional by disabling it. Maybe I did a bad job at describing the issue by starting with my use-case instead of the desired functionality.

It'd be great if we were able to have something like this:

```
- selinux: state=disabled runtime_state=permissive
```

That way I could fulfill my use-case by setting the config to disabled to take full effect on the next boot but until then set it to permissive.

Does that make sense?
